test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -164,8 +164,14 @@ UcxConnection* UcxContext::connect(const struct sockaddr* saddr, size_t addrlen)
 
 void UcxContext::progress()
 {
-    ucp_worker_progress(_worker);
-    progress_io_message();
+    for (unsigned i = 0; i < 1000; ++i) {
+        bool progressed = ucp_worker_progress(_worker);
+        progress_io_message();
+        if (!progressed) {
```

Comment:
but if we send/recv a lot of messages, we don't want the cose of gettimeofday() for every loop

Reply:
if we send/recv a lot of messages, progress always should return > 0 and `gettimeofday` will be called 1/1000 times as before

Reply:
ok, so under stress there should be not so many calls , but i'd rather not rely on it
i think it would be better to keep it as before, and only have for(1000) loop on server side

Reply:
and call `gettimeofday` on client every time?

Reply:
but before the change if (count < 1000) check was only on client...